### PR TITLE
Add multi-database support to HTTP driver

### LINKED ISSUE #227: [4.0] Support database configuration for HTTP driver
Happy to send a PR.

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #227

### CODE REVIEW

**Comment by fbiville** [comment]
I need to figure out why a bunch of tests broke when I pulled extra dependencies before marking this PR ready for review.

**Comment by fbiville** [comment]
Just tested with Liquigraph. Works like a charm.

File: neo4j-jdbc-http/src/main/java/org/neo4j/jdbc/http/driver/CypherExecutor.java [hunk 1]
```diff
@@ -377,6 +386,10 @@ public Integer getOpenTransactionId() {
 		return getTransactionId(this.currentTransactionUrl);
 	}
 
+	private String transactionPath(String databaseName) {
+		return String.format(DB_DATA_TRANSACTION_TEMPLATE, databaseName);
```
Comment (fbiville) [inline neo4j-jdbc-http/src/main/java/org/neo4j/jdbc/http/driver/CypherExecutor.java:390]: it appears there is no need for escaping, as database names can only be made of: `"simple ascii characters, numbers, dots and dashes"` and [these characters are not URI special characters](https://tools.ietf.org/html/rfc3986#section-2.3).

**Comment by conker84** [comment]
LGTM if @jexp has no comments we can merge it.
Thanks a lot @fbiville !

**Comment by fbiville** [comment]
Would it be possible to get a release going once this is merged? Molte grazie!

**Pull request merged**